### PR TITLE
Check if ok when casting Scan src to string

### PR DESCRIPTION
**Pull request by theFong**

### CODE REVIEW

**Comment by bojanz** [comment]
Thanks!

In what case would this happen? Can we add a test for it?

**Comment by theFong** [comment]
I'm using this library here: https://github.com/jinzhu/copier and it will try to set fields differently if the object is a sql.Scanner. Proper error response appears important.

https://github.com/jinzhu/copier/blob/master/copier.go#L625

Will add a test!

**Comment by theFong** [comment]
Added test to cover new error case!

File: amount.go [hunk 1]
```diff
@@ -360,7 +360,10 @@ func (a Amount) Value() (driver.Value, error) {
 // Allows scanning amounts from a PostgreSQL composite type.
 func (a *Amount) Scan(src interface{}) error {
 	// Wire format: "(9.99,USD)".
-	input := src.(string)
+	input, ok := src.(string)
+	if !ok {
+		return fmt.Errorf("invalid src: %v", src)
```
Comment (bojanz) [inline amount.go:365]: This error does not tell us why the value is invalid.

What if we did this instead:
```
return fmt.Errorf("value is not a string: %v", src)
```

**Comment by bojanz** [comment]
@theFong 
I apologize for losing track of your MR, work got a bit busy. I left one last comment for a possible improvement, then we can merge this.

**Comment by bojanz** [comment]
Tweaked and committed. Cheers!

**Pull request closed without merging**